packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -313,6 +321,120 @@ describeNoCompat("Summarizer with local data stores", (getTestObjectProvider) =>
+			},
+		],
+		async () => {
+			// Wait for 100 ms for pending ops to be saved.
+			const pendingOpsTimeout = 100;
+			settings["Fluid.ContainerRuntime.SubmitSummary.waitForPendingOpsTimeout"] =
+				pendingOpsTimeout;
+			const mockLogger = new MockLogger();
+			const container1 = await provider.makeTestContainer();
+			const { summarizer, container: summarizerContainer } = await createSummarizer(
+				provider,
+				container1,
+				{ loaderProps: { configProvider: mockConfigProvider(settings) } },
+				undefined /* summaryVersion */,
+				mockLogger,
+			);
+
+			const defaultDataStore1 = await requestFluidObject<ITestDataObject>(
+				summarizerContainer,
+				"default",
+			);
+
+			// Pause op processing and send ops so there are pending ops in the summarizer.
+			const pendingOpCount = 10;
+			await provider.opProcessingController.pauseProcessing(summarizerContainer);
+			for (let i = 0; i < pendingOpCount; i++) {
+				defaultDataStore1._root.set(`key${i}`, `value${i}`);
+			}
+
+			await assert.rejects(
+				async () => {
+					await summarizeNow(summarizer);
+				},
+				(error: any) => {
+					// The summary should have failed because of "PendingOpsWhileSummarizing" error in "base" stage.
+					return (
+						error.message === "PendingOpsWhileSummarizing" &&
+						error.data.stage === "base"
+					);
+				},
+				"expected PendingOpsWhileSummarizing",
+			);
+
+			// We should have received a PendingOpsWhileSummarizing event with all the pending ops not saved.
+			mockLogger.assertMatch([
+				{
+					eventName: "fluid:telemetry:Summarizer:Running:PendingOpsWhileSummarizing",
+					saved: false,
+					countBefore: pendingOpCount,
+					countAfter: pendingOpCount,
+					timeout: pendingOpsTimeout,
+				},
+			]);
+		},
+	);
+
+	itExpects(
+		"ValidateSummaryBeforeUpload = true. Summary should fail if ops are sent during summarize",
+		[
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel",
+				clientType: "noninteractive/summarizer",
+				error: "PendingOpsWhileSummarizing",
+				beforeGenerate: false,
+			},
+		],
+		async () => {
+			const mockLogger = new MockLogger();
+			const container = await createContainer(provider);
+			await waitForContainerConnection(container);
+
+			const rootDataObject = await requestFluidObject<RootTestDataObject>(container, "/");
+			const dataObject2 = await dataStoreFactory2.createInstance(
+				rootDataObject.containerRuntime,
+			);
+			rootDataObject._root.set("dataStore2", dataObject2.handle);
+			dataObject2._root.set("op", "value");
+
+			const { summarizer } = await createSummarizerFromFactory(
+				provider,
+				container,
+				rootDataObjectFactory,
+				undefined /* summaryVersion */,
+				undefined /* containerRuntimeFactoryType */,
+				registryStoreEntries,
+				mockLogger,
+				mockConfigProvider(settings),
+			);
+			await provider.ensureSynchronized();
+
+			await assert.rejects(
+				async () => {
+					await summarizeNow(summarizer);
```

Comment:
How is this test exercising "ops are sent during summarize"?

Reply:
dataObject2 does that when summarizing by using `mixinSummaryHandler`. I will add a comment above to mention this.